webserver/views/api/v1/datasets.py

```diff
@@ -194,7 +224,22 @@ def delete_class(dataset_id):
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    ds = get_check_dataset(dataset_id)
+    if ds["author"] != current_user.id:
+        raise api_exceptions.APIUnauthorized("You can't delete this class.")
+    class_dict = request.get_json()
+    if "name" not in class_dict:
+        raise api_exceptions.APIBadRequest("name key missing in JSON request.")
+    if class_dict["name"] not in (classes["name"] for classes in ds["classes"]):
```

Comment:
This is similar to the check in the the `add_class` method. @gentlecat 

Here we could raise an error, as you do, or just silently skip the request. There is no HTTP 2xx code which says that something was removed, so we can't use that as an indicator.